src/components/EditorEasyMDE.vue

```diff
@@ -178,14 +223,69 @@ export default {
 					url: generateUrl('apps/notes') + '/notes/' + id + '/attachment',
 					data,
 				})
-				const name = response.data[0].filename
+				console.log("ERROR: Response currently does not return filename")
+				console.log(response)
 				const position = {
 					line: cursor.line,
 				}
 				doc.replaceRange('![' + name + '](' + name + ')', position)
 			}
 			temporaryInput.click()
 		},
+
+		insertText(content) {
+			const doc = this.mde.codemirror.getDoc()
+			const cursor = this.mde.codemirror.getCursor()
+			const position = {
+				line: cursor.line
+			}
+			doc.replaceRange(content, position)
+		},
+
+		surroundText(content) {
+			const doc = this.mde.codemirror.getDoc()
+			const cursorStart = this.mde.codemirror.getCursor('from')
+			const cursorEnd = this.mde.codemirror.getCursor('to')
+			const originalText = doc.getRange(cursorStart, cursorEnd)
+			doc.replaceRange(content + originalText + content, cursorStart, cursorEnd)
+		},
+
+		insertLink() {
+			this.insertText('[title](url)')
+		},
+
+		insertCheckbox() {
+			this.insertText('- [ ] ')
+		},
+
+		makeBold() {
+			this.surroundText('**')
+		},
+
+		makeItalic() {
+			this.surroundText('_')
+		},
+
+		makeMonospace() {
+			this.surroundText('`')
+		},
+
+		makeStrikethrough() {
+			this.surroundText('~~')
+		},
+
+		undo() {
+			// the first "undo" is the editor beeing populated. Do not undo that!
+			if(this.mde.codemirror.historySize().undo>1){
+				this.mde.codemirror.undo()
+			}
```

Comment:
```suggestion
			this.mde.codemirror.undo()
```

This `if` is not needed anymore, since  #825 fixes the original issue! Thanks for pointing me on this!